File: Project 1 Phase 4/meam620-kmel/meam620-kmel/QC-12.12.12/interfaces/kQuadIface-12.12.12/testSendQuadCmd3.m
%Copyright KMel Robotics 2012. Must read KMEL_LICENSE.pdf for terms and conditions before use.
function testSendQuadCmd3()

%open the interface
KQUAD.dev    = '/dev/kbee1';
KQUAD.driver = @kQuadInterfaceAPI;
KQUAD.baud   = 1000000;
KQUAD.id     = 1;
KQUAD.chan   = 1;
KQUAD.type   = 1; %0 for standard, 1 for nano

KQUAD.driver('connect',KQUAD.dev,KQUAD.baud)

while(1)
  
  thrust = 1; %grams. 1 for idle
  roll   = 0; %radians
  pitch  = 0; %radians
  yaw    = 0; %radians
  
  trpy = [thrust roll pitch yaw];
  
  drpy = [0 0 0];     %angular rates (rad/sec)
  kp   = 0*[400 400 470];     %proportional gains
  kd   = 0*[30 30 50];     %derivative gains

  KQUAD.driver('SendQuadCmd3',KQUAD.id, KQUAD.chan, KQUAD.type, ...
               trpy, drpy, kp, kd);

  pause(0.03);
end

File: Project 1 Phase 4/meam620-kmel/meam620-kmel/QC-12.12.12/interfaces/kQuadIface-12.12.12/testSendQuadCmd3.cc
/* Copyright KMel Robotics 2012. Must read KMEL_LICENSE.pdf for terms and conditions before use. */
#include "kQuadInterface.hh"

int main()
{
  char * dev = (char*)"/dev/ttyUSB0";
  int baud   = 921600;
  
  kQuadInterface kqi;
  
  if (kqi.Connect(dev,baud))
  {
    printf("could not connect to the device\n");
    return -1;
  }
  
  if (kqi.StartSendThread())
  {
    printf("could not start thread\n");
    return -1;
  }
  
  uint8_t quadId   = 11;
  uint8_t quadType = 0; //standard
  uint8_t channel  = 1;
  
  float thrust = 150; //(grams) 1 for idle
  float roll   = 0; //radians
  float pitch  = 0; //radians
  float yaw    = 0; //radians
  
  float droll  = 0;
  float dpitch = 0;
  float dyaw   = 0;
  
  float kp_roll   = 950;
  float kp_pitch  = 950;
  float kp_yaw    = 1000;
  
  float kd_roll   = 100;
  float kd_pitch  = 100;
  float kd_yaw    = 100;
  
  while(1)
  {
    kqi.SendQuadCmd3(quadId, quadType, channel, thrust, roll, pitch, yaw,
                     droll,dpitch,dyaw, kp_roll,kp_pitch,kp_yaw, kd_roll,kd_pitch,kd_yaw);
    
    usleep(10000);
    
    printf("."); fflush(stdout);
  }
  
 
  return 0;
}